contracts/libraries/Validator.sol

```diff
@@ -72,4 +31,83 @@ library Validator {
         }
         return computedHash == rootHash;
     }
+
+    // @param txHash      transaction hash
+    // @param rootHash    block header of the merkle tree
+    // @param input0      indicator for nonzero first input
+    // @param input1      indicator for nonzero second input
+    // @param sigs        transaction signatures
+    function checkSigs(bytes32 txHash, bytes32 confirmationHash, bool input0, bool input1, bytes sigs, bytes confirmSignatures)
+        internal
+        pure
+        returns (bool)
+    {
+        require(sigs.length == 130, "two transaction signatures required");
+        require(confirmSignatures.length == 130, "two confirm signatures required");
+        bytes memory sig0 = slice(sigs, 0, 65);
+        bytes memory sig1 = slice(sigs, 65, 65);
+        bytes memory confSig0 = slice(confirmSignatures, 0, 65);
+        
+        bool check0 = true;
+        bool check1 = true;
+        if (input0) {
```

Comment:
If deposit sigs are no longer checked than a tx should always have a first input and this if statement isn't necessary. Are checkSigs being used for deposits?

Reply:
`checkSigs` is not used for deposits b/c there are not confirmsig for it. nothing is wrong with constructing a transaction that uses the second input instead of the first.